spanner: fixed applying default_labels changes

rickard-von-essen opened the pull request:
Before this fix just changing `default_labels` on the provider for a `google_spanner_instance` resulted in this error:

```
googleapi: Error 400: Invalid UpdateInstance request.
[...]
Must specify a non-empty field mask
```

Fixes https://github.com/hashicorp/terraform-provider-google/issues/18445

Added one acceptance tests to surface this issue.

- TestAccSpannerInstance_basicUpdateWithProviderDefaultLabels

```release-note: bug
spanner: fixed issue with applying changes in provider `default_labels` on `google_spanner_instance`
```

NickElliot commented on mmv1/templates/terraform/encoders/spanner_instance_update.go.tmpl:
```diff
@@ -21,6 +21,9 @@ if d.HasChange("display_name") {
 if d.HasChange("labels") {
 	updateMask = append(updateMask, "labels")
 }
+if d.HasChange("terraform_labels") {
+	updateMask = append(updateMask, "labels")
```
these should probably be changed to use an OR tag, otherwise theoretically labels could be in the update mask twice.

rickard-von-essen replied:
Resolved here https://github.com/GoogleCloudPlatform/magic-modules/pull/13269/files#diff-685289a680858e6047eb1bff71bc728ee8bdc56accdbaebe10091815e169f912R21

NickElliot reviewed (changes requested):
one point stood out to me

rickard-von-essen commented:
Addressed the review comment and re-executed the acceptance test.

```
$ make testacc TEST=./google/services/spanner TESTARGS='-run=TestAccSpannerInstance_basicUpdateWithProviderDefaultLabels$$'
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google/services/spanner -v -run=TestAccSpannerInstance_basicUpdateWithProviderDefaultLabels$ -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google/version.ProviderVersion=acc"
=== RUN   TestAccSpannerInstance_basicUpdateWithProviderDefaultLabels
=== PAUSE TestAccSpannerInstance_basicUpdateWithProviderDefaultLabels
=== CONT  TestAccSpannerInstance_basicUpdateWithProviderDefaultLabels
--- PASS: TestAccSpannerInstance_basicUpdateWithProviderDefaultLabels (60.45s)
PASS
ok      github.com/hashicorp/terraform-provider-google/google/services/spanner  61.089s
```

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-13269-old..auto-pr-13269) ( 2 files changed, 66 insertions(+), 1 deletion(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-13269-old..auto-pr-13269) ( 2 files changed, 66 insertions(+), 1 deletion(-))

modular-magician commented:
#### Tests analytics
Total tests: 43
Passed tests: 37
Skipped tests: 5
Affected tests: 1

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>spanner</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccSpannerInstance_basicUpdateWithProviderDefaultLabels</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/develop/test/test/)

modular-magician commented:
🟢 Tests passed during RECORDING mode:
`TestAccSpannerInstance_basicUpdateWithProviderDefaultLabels` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-13269/artifacts/c123b9b2-e9e6-4cb8-802e-8667cb4ae8cf/recording/TestAccSpannerInstance_basicUpdateWithProviderDefaultLabels.log)]

🟢 No issues found for passed tests after REPLAYING rerun.

---






🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-13269/artifacts/c123b9b2-e9e6-4cb8-802e-8667cb4ae8cf/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-13269/artifacts/c123b9b2-e9e6-4cb8-802e-8667cb4ae8cf/recording) for each test
